### PR TITLE
Null correction for allowing multiple operations

### LINKED ISSUE #100: Updating null values handler
Till now, the null values handler was exclusively allowing to select and apply that one method on all columns. If I want to drop a particular column, fill some other with mean and drop null records in another column, it wasn't possible. So the update is that the methods now work on the column list given to each method. *For fill_missing parameter, if column list is given as empty list then the method(mean/median) is applied everywhere.
There is no option to select the drop rows method. After the user chosen methods are done, the drop rows is run by default to clear any null rows the user may have missed or specifically kept to be removed in this manner.

Example - Do the following => Col A: mean, Col B: drop the column, Col C: fill with a value. Can be done only after the update in the PR that will be linked with this issue.

### PR DESCRIPTION
**Pull request by Priyesh1202**
This PR is a fix for the issue 100. It updates the code such that multiple null values handling operations can take place . 

https://github.com/preprocessy/preprocessy/issues/100

- Fixed #100 

Checklist:

- [x] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [x] Add or update relevant docs, in the docs folder and in code.
- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [ ] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest`, no tests failed.

### CODE REVIEW

File: tests/test_null.py [hunk 1]
```diff
@@ -26,45 +27,47 @@ def test_null_dataframe():
...
+#             "train_df": dataframe1,
+#             "fill_missing": [[],"mean"],
+#             "fill_values": {"Test": "Tata"},
+#         },
+#     ],
+# )
+# def test_multiple_args(test_input):
+#     with not pytest.raises(ArgumentsError):
+#         handler = NullValuesHandler()
+#         handler.execute(params=test_input)
```
Comment (Saif807380) [inline tests/test_null.py:52]: can be removed

**Review by Saif807380 — CHANGES_REQUESTED** [review]

File: docs/parameters.rst [hunk 2]
```diff
@@ -18,14 +18,12 @@ Any parameter marked ``*`` are essential to the smooth working of the pipeline a
 
 - replace_cat_nulls(``int``): Integer value that is used to fill all null cells in categorical columns(taken from ``cat_cols`` parameter)
 
-- drop(``bool``): ``True`` if all records with null values to be removed. ``False`` otherwise. If ``fill_missing`` and ``fill_values`` and ``drop`` are not given then ``drop`` is set to True.
+- drop_cols(``list``): List of column names to be dropped.
 
-- fill_missing(``str``): ``mean`` if null values to be filled with mean of that column, ``median`` otherwise.
+- fill_missing(``list``): List with the format [[columns that the method will be applied on], method]. For example: [["Age"],"mean"].
```
[the remark this answers is not in the document]
Comment (Parth576) [reply docs/parameters.rst:23]: It won't work like that. The columns given in ```fill_missing``` can be filled by either mean or median.

[the remark this answers is not in the document]
Comment (Priyesh1202) [reply docs/parameters.rst:23]: You want me to add for that too? I can do that actually won't be too much of a change. Just a few changes here and there. 
Proposed solution: format: [method,col_list,..] so if col A -> mean and col B-> median then ["mean","col A","median","col B"]. So ill check for the keywords(mean/median) and will take the element next to the keyword match to extract its col_list. Will this be alright?

[the remark this answers is not in the document]
Comment (Parth576) [reply docs/parameters.rst:23]: If there are many columns and the user wants to fill them with mean - ```["mean", "col A", "mean", "col B", "mean", "col 2"]```. Does not look that intuitive.

[the remark this answers is not in the document]
Comment (Priyesh1202) [reply docs/parameters.rst:23]: cols will be list sorry

[the remark this answers is not in the document]
Comment (Priyesh1202) [reply docs/parameters.rst:23]: So like the parameter can be of max length 4.. like [method, col list to apply, method, col list to apply]

[the remark this answers is not in the document]
Comment (Parth576) [reply docs/parameters.rst:23]: Won't a dictionary be better than this so we can easily check the key? 
Example 1 - mean and median ```{ 'mean' : ['col A', 'col B'], 'median' : ['col C'] }```
Example 2 - only mean ```{ 'mean' : ['col A', 'col B'] }```

[the remark this answers is not in the document]
Comment (Saif807380) [reply docs/parameters.rst:23]: A dictionary will be a good option. Or we can just have a list of lists, where the 0th index is for list of cols to be filled by mean and the next index is list of cols to filled by median.

[the remark this answers is not in the document]
Comment (Priyesh1202) [reply docs/parameters.rst:23]: List of list will be problematic is only median is required.. Ill make it a dictionary then..

**Review by Saif807380 — APPROVED** [review]
The code in `__validate_input()` has become too complicated now. Most of it just type checks across all modules. We should look into type safety options in python to simplify all of it.

**Pull request merged**